src/swoop/api/main.py

```diff
@@ -14,6 +15,20 @@
 
 app: FastAPI = FastAPI()
 
+app.state.settings = settings
+
+@app.on_event("startup")
+async def startup_event():
+    """Connect to database on startup."""
+    pools = await connect_to_db(settings)
```

Comment:
Seems like this should be `app.state.settings`.

Reply:
Maybe consider this too:

```suggestion
    app.state.readpool, app.state.writepool = await connect_to_db(app.state.settings)
```

Reply:
that's much nicer 👍